jekyll/_cci2/local-cli.md

```diff
@@ -7,170 +7,311 @@ categories: [troubleshooting]
 
-### Checking for available options
+### Running a Job
 {:.no_toc}
 
-As we update the local tool regularly, there may be options for a command that aren't documented here.  You can check for these options with `circleci help <command>`.
+The CLI allows you to run a single job from CircleCI on your desktop using Docker. 
 
-## Limitations
+```sh
+$ circleci local execute --job JOB_NAME
+```
+
+Let's run an example build on our local machine on one of CircleCI's demo applications:
+
+```sh
+git clone https://github.com/CircleCI-Public/circleci-demo-go.git
+cd circleci-demo-go
+circleci local execute --job build
+```
+
+The commands above will run the entire _build_ job (only jobs, not workflows, can be run locally). The CLI will use Docker to pull down the requirements for the build and will then execute your CI steps locally. In this case, Golang and Postgres docker images are pulled down, allowing the build to install dependencies, run the unit tests, test the service is running and so on.
 
-Although running jobs locally with `circleci` is very helpful, there are some limitations.
 
-### Machine Executor
+### Limitations of Running Jobs Locally
 {:.no_toc}
 
+Although running jobs locally with `circleci` is very helpful, there are some limitations.
+
+**Machine Executor**
+
 You cannot use the machine executor in local jobs. This is because the machine executor requires an extra VM to run its jobs.
 
-### Caching
-{:.no_toc}
+**Workflows**
 
-Caching is not currently supported in local jobs. When you have either a `save_cache` or `restore_cache` step in your config, `circleci` will skip them and display a warning.
+The CLI tool does not provide support for running workflows. By nature, workflows leverage running builds in parallel on multiple machines allowing you to achieve faster, more complex builds. Because the CLI is only running on your machine, it can only run single **jobs** (which make up parts of a workflow).
 
-### Environment Variables
-{:.no_toc}
+**Caching and Online-only Commands**
+
+Caching is not currently supported in local jobs. When you have either a `save_cache` or `restore_cache` step in your config, `circleci` will skip them and display a warning.
 
-For security reasons, encrypted environment variables configured in the UI will not be exported into local builds. As an alternative, you can specify env vars to the CLI with the `-e` flag. See the output of `circleci help build` for more information. If you have multiple environment variables, you must use the flag for each variable, for example, `circleci build -e VAR1=FOO -e VAR2=BAR`.
+Further, not all commands may work on your local machine as they do online. For example, the Golang build reference above runs a `store_artifacts` step, however, local builds won't upload artifacts. If a step is not available on a local build you will see an error in the console.
```

Comment:
can we add what that error would be? Also is there something we should add about where the files are saved locally? in /project/xyz 
From that twitch user video, where 1) the guy couldn't find his files and 2) also was failing because he needed to add a sleep to access the container

Reply:
I believe this is the error that gets displayed:

> Error: FAILED with error not supported.

I brought this up with @zzak, suggesting maybe we should leave that out of the docs as it's not particularly explicit of an error message. Thoughts?

Reply:
@teesloane, @michelle-luna: Sorry I'm just seeing this PR now.

Here is the error for restore cache:

```
====>> Restoring Cache
Error: 
Skipping cache - error checking storage: not supported


Step failed
```

For save cache:

```
====>> Saving Cache
Error: 
Skipping cache - error checking storage: not supported

Step failed
```

And for store artifacts, you can see.. not very helpful!

```
====>> Uploading artifacts
Uploading /home/circleci/project/coverage.txt to coverage.txt
Uploading /home/circleci/project/coverage.txt (53 kB): Error: FAILED with error not supported
```